src/main/java/org/damap/base/conversion/AbstractTemplateExportSetup.java

```diff
@@ -53,18 +53,27 @@ protected void exportSetup(long dmpId) {
     costList = dmp.getCosts();
 
     // determine project leader/coordinator/principal investigator
-    Optional<Contributor> projectLeaderOpt =
+    List<Contributor> projectLeaderOpt =
         dmp.getContributorList().stream()
             .filter(
-                contributor -> contributor.getContributorRole() == EContributorRole.PROJECT_LEADER)
-            .findFirst();
-    if (projectLeaderOpt.isPresent())
-      projectCoordinator =
-          ContributorDOMapper.mapEntityToDO(projectLeaderOpt.get(), new ContributorDO());
-    else
+                contributor ->
+                    contributor.getContributorRole() == EContributorRole.PROJECT_LEADER
+                        || contributor.getContributorRole() == EContributorRole.PROJECT_COORDINATOR
+                        || contributor.getContributorRole()
+                            == EContributorRole.PRINCIPAL_INVESTIGATOR)
+            .toList();
+
+    if (!projectLeaderOpt.isEmpty()) {
```

Comment:
Java streams .toList() already return an empty list when no element was found, there is no reason for this intermediate list.

Reply:
Done